### PR TITLE
mtest: Fix spawn tests after return value patch

### PR DESCRIPTION
**Pull request by wesbland**
The return value patch merged as part of pmodels/mpich#2640 was missing
some code for spawn tests where the spawned processes would call
`MPI_Finalize`. This caused all of those tests to hang.

Add the appropriate calls here to make sure the tests complete properly.

### CODE REVIEW

**Comment by wesbland** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most

File: test/mpi/cxx/spawn/spawnargvx.cxx [hunk 1]
```diff
@@ -117,6 +117,8 @@ int main( int argc, char *argv[] )
     /* Note that the MTest_Finalize get errs only over COMM_WORLD */
     if (parentcomm == MPI::COMM_NULL) {
 	MTest_Finalize( errs );
+    } else {
+        MPI_Finalize();
```
Comment (pavanbalaji) [inline test/mpi/cxx/spawn/spawnargvx.cxx]: Is the indentation correct here?

File: test/mpi/cxx/spawn/spawnintrax.cxx [hunk 2]
```diff
@@ -203,6 +203,8 @@ int main( int argc, char *argv[] )
        if both call MTest_Finalize */
     if (parentcomm == MPI::COMM_NULL) {
 	MTest_Finalize( errs );
+    } else {
+        MPI_Finalize();
```
Comment (pavanbalaji) [inline test/mpi/cxx/spawn/spawnintrax.cxx]: Indentation?

File: test/mpi/cxx/spawn/spawnargvx.cxx (hunk 1 above)
Comment (wesbland) [reply test/mpi/cxx/spawn/spawnargvx.cxx]: Yeah, the C++ tests seem to have escaped the previous whitespace cleanups. I can add another patch to fix the whitespace throughout if you think it's fine (we've done that with the test suite previously).

**Comment by wesbland** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by pavanbalaji** [comment]
I'm not going to review the white space patch.  😄 

But I'll approve this patch.

**Comment by wesbland** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by wesbland** [comment]
Missed `f90/datatype/structf`. This fixes that one too.

test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by wesbland** [comment]
test:jenkins/ch4/ofi

**Comment by wesbland** [comment]
test:jenkins/ch4/ofi

**Comment by wesbland** [comment]
test:ch4/ofi/sockets/gnu/force-am-disabled/shm-exclusive/debug

**Pull request merged**